code/game/machinery/constructable_frame.dm

```diff
@@ -685,6 +685,33 @@ obj/item/weapon/circuitboard/rdserver
 							"/obj/item/weapon/stock_parts/scanning_module" = 1,
 							"/obj/item/weapon/stock_parts/console_screen" = 2)
 
+
+
+/obj/item/weapon/circuitboard/smartfridge/attackby(var/obj/item/weapon/G, var/mob/user)
+	if(issolder(G))
+		var/obj/item/weapon/solder/S = G
+		var/list/static/smartfridge_choices = list("Food smartfridge" = /obj/item/weapon/circuitboard/smartfridge/,
+											"Secure chemistry smartfridge" = /obj/item/weapon/circuitboard/smartfridge/medbay,
```

Comment:
Very well, doing so now.

Does the static list help in any way? Noticed you recommending people to use it